includes/Checker/Checks/Plugin_Header_Text_Domain_Check.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\Plugin_Header_Text_Domain_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use Exception;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Static_Check;
+
+/**
+ * Check for plugin header text domain.
+ *
+ * @since n.e.x.t
+ */
+class Plugin_Header_Text_Domain_Check implements Static_Check {
+
+	/**
+	 * Amends the given result by running the check on the associated plugin.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of
+	 *                   the check).
+	 */
+	public function run( Check_Result $result ) {
+
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		$plugin_main_file = $result->plugin()->abspath();
```

Comment:
See below.
```suggestion
		$plugin_main_file = WP_PLUGIN_DIR . '/' . $result->plugin()->basename();
```

Reply:
Could we use `$result->plugin()->location();`

Reply:
@spacedmonkey, it seems that the function only returns the absolute path of the plugin folder, but we actually need the path to the plugin's main file.